src/modules/reporting/components/reporting-dispute-form/reporting-dispute-form.jsx

```diff
@@ -64,18 +63,19 @@ export default class ReportingDisputeForm extends Component {
   }
 
   componentWillMount() {
-    this.getMarketDisputeInfo()
+    this.updateDisptueInfoState()
     if (this.props.accountDisputeData) {
       this.setAccountDisputeData(this.props.accountDisputeData)
     }
   }
 
   shouldComponentUpdate(nextProps, nextState) {
+    if (nextProps.availableRep !== this.props.availableRep) return true
     return !isEqual(nextState, this.state)
   }
 
-  componentDidUpdate() {
-    this.getMarketDisputeInfo()
+  componentDidUpdate(nextProps) {
+    this.updateDisptueInfoState()
```

Comment:
This isn't passing LGTM, it's an anti-pattern because it can cause state refreshes to cycle infinitely.